File: ui/piquick_objects/lib/model/render_model.dart
class RenderModel {
  int numImages;
  bool azimuthAug;
  bool elevationAug;
  int resolution;
  bool modeMulti;
  bool modeStatic;
  bool modeFrontView;
  bool modeFourView;
  String engine;
  bool onlyNorthernHemisphere;

  RenderModel({
    required this.numImages,
    required this.azimuthAug,
    required this.elevationAug,
    required this.resolution,
    required this.modeMulti,
    required this.modeStatic,
    required this.modeFrontView,
    required this.modeFourView,
    required this.engine,
    required this.onlyNorthernHemisphere,
  });

  void updateSettings({
    required int numImages,
    required bool azimuthAug,
    required bool elevationAug,
    required int resolution,
    required bool modeMulti,
    required bool modeStatic,
    required bool modeFrontView,
    required bool modeFourView,
    required String engine,
    required bool onlyNorthernHemisphere,
  }) {
    this.numImages = numImages;
    this.azimuthAug = azimuthAug;
    this.elevationAug = elevationAug;
    this.resolution = resolution;
    this.modeMulti = modeMulti;
    this.modeStatic = modeStatic;
    this.modeFrontView = modeFrontView;
    this.modeFourView = modeFourView;
    this.engine = engine;
    this.onlyNorthernHemisphere = onlyNorthernHemisphere;
  }
}

File: backend/models/render_model.py
from typing import Dict, List, Optional
"""
This module defines data models for rendering 3D object groups using Pydantic's BaseModel. 
It includes settings for rendering and grouping of 3D objects.

Classes:
    RenderSettings: Represents the configuration for rendering 3D object groups, 
                    including parameters like number of images, resolution, and augmentation options.
    Group: Represents a group of 3D objects with associated rendering settings.
"""
from pydantic import BaseModel, Field

class RenderSettings(BaseModel):
    """Settings for rendering object groups"""
    num_images: int = Field(default=12)
    azimuth_aug: bool = Field(default=True)
    elevation_aug: bool = Field(default=False)
    resolution: int = Field(default=256)
    mode_multi: bool = Field(default=True)
    mode_front_view: bool = Field(default=False)
    mode_four_view: bool = Field(default=False)
    only_northern_hemisphere: bool = Field(default=True)


class Group(BaseModel):
    """Group of 3D objects with rendering settings"""
    name: str
    object_ids: List[str]
    settings: RenderSettings = Field(default_factory=RenderSettings)